backend/data_classes.py

```diff
@@ -0,0 +1,167 @@
+    :type phone_number: str
+    :param translated_name: Translated name of the reporter to English
+    :type translated_name: str
+    :param phone_number_2: Second phone number of the reporter
+    :type phone_number_2: str
+    :param email: Email of the reporter
+    :type email: Optional[str]
+
+    :return: None
+    :rtype: None
+    """
+
+    #  if you are wondering why some args are out of order, @charitarthchugh changed because fields without default values should appear after fields with default values
+    first_name: str
+    last_name: str
+    relationship: str
+    date_of_birth: datetime
+    nationality: str
+    country_residence: str
+    phone_number: str
+    translated_name: Optional[str] = None
+    phone_number_2: Optional[str] = None
+    email: Optional[str] = None
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]):
+        """Generate a Reporter class from a dictionary
+
+        :param d: Dictionary with the named parameters to init Reporter as keys
+        :type d: Dict[str, Any]
+        :return: a Reporter class with
+        :rtype: _type_
+        """
+        return Reporter(**d)
+
+
+@dataclass
+class Child:
+    """
+    Dataclass for information about reported
+
+    :param first_name: Child's first name
+    :type
+    :param last_name: Child's last name
+    :param  translated_name: Child's Tranaslated name to english
+    :param date_birth:  Date of Birth of Child in python's datetime format
+    :param gender: male or female only
+    :param nationality: Nationality of the missing child
+    :param blood _type:  blood type of child
+    :param reported:  .... forgot what we doing here
+    :param last_seen: Dict of locations where the child was last seen
+    :param comment: optional description of child (optional)
+    :param picture: picture of child (Optional)
+    """
+
+    first_name: str
+    last_name: str
+    date_of_birth: datetime
+    gender: str
+    nationality: str
+    reported: str
+    last_seen: Dict[str, Any]
+    translated_name: Optional[str] = None
+    blood_type: Optional[str] = None
+    comment: Optional[str] = None
+    picture: Optional[Any] = None
+
+    @staticmethod
+    def from_dict(d: Dict[str, Any]):
+        current_date = pytz.utc.localize(datetime.today())
+        birth_date = pytz.utc.localize(datetime.strptime(d["date_birth"], "%Y.%m.%d"))
+
+        # Check valid gender
+        if not (d["gender"] == "male" or d["gender"] == "female"):
+            raise ValueError("No valid gender specified")
+
+        # Check valid blood type
+        if "blood_type" in d:
+            if not d["blood_type"] in __blood_types:
+                raise ValueError("Invalid blood type specified")
+
+        # Check last_seen dionary
+        if (
+            "date" in d["last_seen"]
+            and "country" in d["last_seen"]
+            and "city" in d["last_seen"]
+        ):
+            lastseen_date = pytz.utc.localize(
```

Comment:
this is why we need to verify city and country. Because you have to localize to the country and not just localize to UTC because it will be displayed wrongly if you don't